rclcpp/include/rclcpp/node.hpp

```diff
@@ -67,22 +67,29 @@ class Node : public std::enable_shared_from_this<Node>
   /// Create a new node with the specified name.
   /**
    * \param[in] node_name Name of the node.
+   * \param[in] name_space Namespace of the node.
    * \param[in] use_intra_process_comms True to use the optimized intra-process communication
    * pipeline to pass messages between nodes in the same process using shared memory.
    */
   RCLCPP_PUBLIC
-  explicit Node(const std::string & node_name, bool use_intra_process_comms = false);
+  explicit Node(
+    const std::string & node_name,
+    const std::string & name_space = "",
```

Comment:
I explained in the other pr that I feel that namespace is an important configurations that might ought to be directly in the signature, but I could be convinced otherwise.

In regard to the option structure, I neither like the idea of exposing the rcl structure directly nor do I like the idea of holding up this pr in order to implement a full option class for node. We do need to do the latter at some point, but I didn't want to do it right now.

The `use_intra_process_comms` is meant to be temporary. The idea is that in the future it would default to on and there would be an option in the options structure to disable them.